lgc/patch/MeshTaskShader.cpp

```diff
@@ -284,19 +373,24 @@ Value *MeshTaskShader::getShaderRingEntryIndex(Function *entryPoint) {
 // @param entryPoint : Entry-point of task shader
 // @returns : The payload ring entry offset of current workgroup
 Value *MeshTaskShader::getPayloadRingEntryOffset(Function *entryPoint) {
-  assert(getShaderStage(entryPoint) == ShaderStageTask); // Must be task shader
-
   if (!m_payloadRingEntryOffset) {
     Value *ringEntryIndex = getShaderRingEntryIndex(entryPoint);
     Value *payloadRingBufDesc = m_pipelineSysValues.get(entryPoint)->getTaskPayloadRingBufDesc();
 
+    // NOTE: Make sure below calculation follows payload ring descriptor getter and is prior to any task payload
+    // access operations.
+    auto savedInsertPoint = m_builder->saveIP();
```

Comment:
You can use an InsertionPointGuard here.

Reply:
Thank you. That is very useful.